DataDonationCryptoObjC/DataDonationCryptoObjC/Sources/InternalServices/DonorKeyHolder.swift

```diff
@@ -17,14 +17,37 @@
 import Data4LifeCrypto
 
 protocol DonorKeyHolderProtocol {
+
+    @discardableResult
+    func generateKeyPair(for programName: String) throws -> KeyPair
+
+    @discardableResult
+    func createKeyPair(from data: Data, for programName: String) throws -> KeyPair
+
+    func fetchKeyPair(for programName: String) throws -> KeyPair
+    func deleteKeyPair(for programName: String) throws
+
     func privateKey(for programName: String) throws -> AsymmetricKey
     func publicKey(for programName: String) throws -> AsymmetricKey
-    func deleteKeyPair(for programName: String) throws
-    func createKeyPair(from data: Data, for programName: String) throws
 }
 
-enum DonorKeyHolderError: Error {
-    case couldNotStoreKeyPair(reason: String)
+
+extension DonorKeyHolderProtocol {
+    func privateKey(for programName: String) throws -> AsymmetricKey {
+        do {
+            return try fetchKeyPair(for: programName).privateKey
+        } catch {
+            return try generateKeyPair(for: programName).privateKey
+        }
+    }
+
+    func publicKey(for programName: String) throws -> AsymmetricKey {
+        do {
+            return try fetchKeyPair(for: programName).publicKey
+        } catch {
+            return try generateKeyPair(for: programName).publicKey
+        }
+    }
```

Comment:
its technically just a convenient method in case we want to fetch a keypair that doesnt exist yet, technically at this point either we stored one (so this method would just fetch it) or we generate it if its not there.

Reply:
For the moment I would feel safer with exceptions. But I leave that up to you...you should at least tag this, in my eyes with a TODO comment of some sort and we are circle back once we have more informations.